lib/instana/agent.rb

```diff
@@ -50,7 +49,7 @@ def initialize
       @thread_spawn_lock = Mutex.new
 
       # Detect platform flags
-      @is_linux = (RUBY_PLATFORM =~ /linux/i) ? true : false
+      @is_linux = RbConfig::CONFIG['host_os'] == 'linux'.freeze
```

Comment:
On a local linux box here, `RbConfig::CONFIG['host_os']` returns "linux-gnu" - can we update this to do a substring search?